python/pyspark/sql/tests/test_arrow.py

```diff
@@ -830,7 +830,7 @@ def test_createDataFrame_empty_partition(self):
         pdf = pd.DataFrame({"c1": [1], "c2": ["string"]})
         df = self.spark.createDataFrame(pdf)
         self.assertEqual([Row(c1=1, c2="string")], df.collect())
-        self.assertGreater(self.spark.sparkContext.defaultParallelism, len(pdf))
+        self.assertGreater(self._legacy_sc.defaultParallelism, len(pdf))
```

Comment:
`self._legacy_sc` is available in both Spark Connect and Spark Classic, so missing `Spark Context` won't be a blocker any more

Reply:
Wait, this won't be available in `pyspark-connect`. Can we skip if `pyspark.util.is_remote_only`?

Reply:
ah, let me see

Reply:
added, please take another look

Reply:
so we have 3 testing mode:
1, spark classic (with jvm access);
2, spark connect (with jvm (`_legacy_sc`) access);
3, spark connect python only (no jvm)